components/match2/wikis/warcraft/match_group_input_custom.lua

```diff
@@ -78,682 +138,418 @@ function CustomMatchGroupInput._readDate(matchArgs)
+---@return table<string, WarcraftParticipant>
+function MapFunctions.getPartyParticipants(mapInput, opponent, opponentIndex)
+	local players = opponent.match2players
+
+	-- resolve the aliases in case they are used
 	local prefix = 't' .. opponentIndex .. 'p'
-	map[prefix .. '1race'] = Logic.emptyOr(map[prefix .. '1race'], map['race' .. opponentIndex])
-	map[prefix .. '1heroes'] = Logic.emptyOr(map[prefix .. '1heroes'], map['heroes' .. opponentIndex])
+	mapInput[prefix .. '1race'] = Logic.emptyOr(mapInput[prefix .. '1race'], mapInput['race' .. opponentIndex])
+	mapInput[prefix .. '1heroes'] = Logic.emptyOr(mapInput[prefix .. '1heroes'], mapInput['heroes' .. opponentIndex])
 
-	for playerIndex, player in pairs(players) do
-		local faction = Logic.emptyOr(map[prefix .. playerIndex .. 'race'], player.extradata.faction, Faction.defaultFaction)
-		faction = Faction.read(faction)
+	local participants = {}
+
+	Array.forEach(players, function(player, playerIndex)
+		local faction = Faction.read(mapInput['t' .. opponentIndex .. 'p' .. playerIndex .. 'race'])
+			or player.extradata.faction
 
 		participants[opponentIndex .. '_' .. playerIndex] = {
-			faction = faction,
+			faction = Faction.read(faction or player.extradata.faction),
 			player = player.name,
-			heroes = CustomMatchGroupInput._readHeroes(
-				map[prefix .. playerIndex .. 'heroes'],
+			heroes = MapFunctions.readHeroes(
+				mapInput[prefix .. playerIndex .. 'heroes'],
 				faction,
 				player.name,
-				Logic.readBool(map[prefix .. playerIndex .. 'heroesNoCheck'])
+				Logic.readBool(mapInput[prefix .. playerIndex .. 'heroesNoCheck'])
 			),
 		}
-	end
+	end)
 
 	return participants
 end
 
----@param players table[]
----@param map table
----@param opponentIndex integer
----@param participants table<string, table>
----@return integer
-function CustomMatchGroupInput._processTeamPlayerMapData(players, map, opponentIndex, participants)
-	local amountOfTbds = 0
-	local playerData = {}
-
-	local numberOfPlayers = 0
-	for prefix, playerInput, playerIndex in Table.iter.pairsByPrefix(map, 't' .. opponentIndex .. 'p') do
-		numberOfPlayers = numberOfPlayers + 1
-		if playerInput:lower() == TBD then
-			amountOfTbds = amountOfTbds + 1
-		else
-			local link = Logic.emptyOr(map[prefix .. 'link'], Variables.varDefault(playerInput .. '_page')) or playerInput
-			link = mw.ext.TeamLiquidIntegration.resolve_redirect(link):gsub(' ', '_')
-
-			playerData[link] = {
-				faction = Faction.read(map[prefix .. 'race']),
-				position = playerIndex,
-				heroes = map[prefix .. 'heroes'],
-				heroesCheckDisabled = Logic.readBool(map[prefix .. 'heroesNoCheck']),
-				playedRandom = Logic.readBool(map[prefix .. 'random']),
-			}
+---@param mapInput table # the input data
+---@param participants table<string, WarcraftParticipant>
+---@param opponents table[]
+---@return string
+function MapFunctions.getMode(mapInput, participants, opponents)
```

Comment:
Isn't this function Opponent.getLegacyMode basically?

Reply:
iirc:
the opp function runs with opp types which are not applicable here
and it handles team vs team a bit different

Reply:
Would it be fine adjusting to use one of the two commons ones?

Reply:
no
the commons ones expect opp types which (inside team matches for example) just do not exist in this context
like the maps are 1v1/2v2/... within a team vs team match

granted for party opp we tech. could use them but as soon as there is a team opp in the match we can not use them
hence we would need an additional check for that and still do the same code as now for that case
which would just increase complexity of the code

Reply:
for legacy purposes alone yes
also it is the only way to condition on the type of game in match2games